scenario06/README.md

```diff
@@ -0,0 +1,123 @@
+# Scenario 6: Deploy cilium, spire and k8s worload registrar
+
+Description of the scenario: the goal of the scenario exposed by the image below is to use the SPIRE [k8s-workload-registrar](https://github.com/spiffe/spire/tree/main/support/k8s/k8s-workload-registrar) component to automatically assign the SPIFFE ID to a workload (nginx) and verify that the ID was created as a k8s label in cilium. This label can be used for policy enforcement using the [Cilium Network Policies](https://docs.cilium.io/en/stable/concepts/kubernetes/policy/#ciliumnetworkpolicy) (CNPs). 
+
+<img src="../imgs/scenario06.png" alt="drawing" width="800"/>
+
+For this tutorial the following steps will be performed:
+1. Deploy two minikube cluster;
+1. Configure Service Accounts in cluster1 to access TokenReview API to be used by `spire-server` on cluster2
+1. Deploy `spire-server` on cluster2
+1. Insert registration entries for `cilium-agent` and `k8s-workload-registrar` in `spire-server`;
+1. Deploy `spire-agent` + `cilium-agent` + `k8s-workload-registrar` + `nginx` on cluster1;
+1. Check that `nginx` SPIFFE ID was automatically created;
+1. Check that `nginx` SPIFFE ID was created as a k8s label in `cilium-agent`
+
+## Software Requirements: 
+
+Be sure that you have already installed: 
+
+- minikube v1.21 + docker driver
+- jq
+- helm3
+
+## Steps
+
+> :memo: The following steps were tested with Manjaro and Ubuntu 20.04 Linux distributions. 
+
+Create clusters:
+
+```
+./0-start-clusters.sh
+```
+
+Deploy the pods in cluster1 and cluster2:
+
+```
+./1-deploy.sh
+```
+
+Check if the registration entries were created in `spire-server`: 
+
+```
+kubectl exec spire-server-0 -n spire -c spire-server --context cluster2 -- ./bin/spire-server entry show  
+```
+
+The output should be: 
+
+```
+Found 4 entries
+Entry ID         : fee71387-3b2d-4b64-9b9a-671b12387c8b
+SPIFFE ID        : spiffe://example.org/ciliumagent
+Parent ID        : spiffe://example.org/spire/agent/k8s_psat/demo-cluster/59092f2c-bb7f-424d-8110-fc6668beaf59
+Revision         : 0
+TTL              : default
+Selector         : unix:uid:0
+Admin            : true
+
+Entry ID         : 1a08e0ad-564e-4c68-bf16-5d96342908af
+SPIFFE ID        : spiffe://example.org/k8s-workload-registrar/demo-cluster/node/cluster1
+Parent ID        : spiffe://example.org/spire/server
+Revision         : 0
+TTL              : default
+Selector         : k8s_psat:agent_node_uid:59092f2c-bb7f-424d-8110-fc6668beaf59
+Selector         : k8s_psat:cluster:demo-cluster
+
+Entry ID         : fd7c4aeb-2724-48f2-98ac-419050f7e84f
+SPIFFE ID        : spiffe://example.org/ns/default/sa/default
```

Comment:
The SPIFFE ID of the workload should be generate for k8s-registrar right? and not static as `spiffe://example.org/ns/default/sa/default`... I was confused about this .... the SPIFFE ID should contains something generated such as pod-uid for example

If the correct SPIFFE ID is `spiffe://example.org/ns/default/sa/default`, the string should not have the part `sa/default` right?

Reply:
As we have talked, this spiffe id format is the default ones generated by k8s workload registrar. It does not contain the pod name. The template can be configured in its configuration file. I decided to removed to keep it simple. If we need to change one day, be aware to escape the go template in `identity_template` k8s workload registrar configuration, because here is a helm template (so helm will try to resolve it during the installation)